### PR TITLE
Restructure document

### PR DESCRIPTION
**Pull request by sareh**
No related ticket. 

**Overall change:** Restructures Document - moves logic from `components/Document` out into different parts of the codebase - `ResourceHints` in `components` and `Document` under `server`. There should be no external change of the page.

**Code changes:**

- Move ResourceHints component so it is no longer nested under Document
- Move Document from components directory to server/Document directory
- Rename to generalise to Document
- Move rendering of Document and collection of assets and styles from `server/index` into the `server/Document/index`

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [x] Test engineer approval
- [x] I have followed [the merging checklist](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/MERGE_PROCESS.md) and this is ready to merge.

### CODE REVIEW

**Review by bcmn — DISMISSED** [review]
Looks good -- I wonder if either this or #1332 should be blocked on the other, though? The overlap between them makes me worry about merge accidents.

File: src/server/Document/index.jsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { renderToString, renderToStaticMarkup } from 'react-dom/server';
+import { ServerStyleSheet } from 'styled-components';
+import { Helmet } from 'react-helmet';
+import { ServerApp } from '../../app/containers/App';
+
+import { getStyleTag } from '../styles';
+import getAssetsArray from '../assets';
+import Document from './component';
```
Comment (dr3) [inline src/server/Document/index.jsx]: Can I be a huge pain and ask for this to be `import DocumentComponent` please?, Document inside Document isnt v pretty imo

**Review by dr3 — CHANGES_REQUESTED** [review]
V nit picky comment, Otherwise looks great 🎉 fab job

File: src/server/Document/index.jsx (hunk 1 above)
Comment (sareh) [reply src/server/Document/index.jsx]: Change has been made.

**Review by dr3 — DISMISSED** [review]

**Review by bcmn — DISMISSED** [review]

**Comment by sareh** [comment]
@bcmn @dr3 Just merged in `latest` & resolved conflicts. If you could please review it again, that'd be great. :)

**Review by dr3 — APPROVED** [review]

**Review by bcmn — APPROVED** [review]

**Comment by jamesbrumpton** [comment]
👍

**Pull request merged**